lib/tokenize.cpp

```diff
@@ -7615,7 +7615,13 @@ void Tokenizer::findGarbageCode() const
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
         if (tok->str() == "{")
             tok = tok->link();
-        else if (tok->isKeyword() && nonGlobalKeywords.count(tok->str()) && !Token::Match(tok->tokAt(-2), "operator %str%"))
+        else if (tok->isKeyword() && Token::simpleMatch(tok->previous(), ") try")) { // function-try-block
+            const Token* endTok = Token::findsimplematch(tok->next(), "catch (");
```

Comment:
findsimplematch could in theory find a catch for a different try. I would suggest we explicitly look up the { and then goto the catch with a link call. Then I assume some while loop should be used to skip all catches.

```
    void f() try {
        try {}
        catch (E1){}
     }
     catch(E2) {}
     catch(E3) {}
```